### DISCUSSION TITLE
Should the methods match?

### DISCUSSION THREAD

**Discussion by diogo-rossi** [haiiliin/abqpy#1188] [Q&A] [answered]
Should the methods in these two classes match?

https://github.com/haiiliin/abqpy/blob/16ed3e4923f44201b78d40ac271beb2476e672d9/src/abaqus/XY/XYSession.py#L20

https://github.com/haiiliin/abqpy/blob/16ed3e4923f44201b78d40ac271beb2476e672d9/src/abaqus/XY/XYData.py#L14

If so, some type annotations needs to be corrected and the docstrings may be modified.

I'm not sure if it is necessary to add implementation on `XYData` class
___

I mean, not all the methods, but only the ones with same name and signature.

**Comment by haiiliin** [comment, marked as answer]
Yes, I think it should match. But I think there are some issues in the current code, method like XYData.. in XYSession should not be staticmethods and methods like XYDataFrom in XYData should be staticmethods.

**Comment by diogo-rossi** [comment]
@haiiliin There is also a missing method `XYDataFromShellThickness` in class `XYSession` that exists in `XYData`.

Should it be add to `XYSession`?

**Reply by haiiliin** [reply]
Yes, I think so, somehow it is missed.

**Reply by diogo-rossi** [reply]
Done on [be45282](https://github.com/haiiliin/abqpy/pull/1190/commits/be452825e092dca65d039795519eab21d2351f41)
